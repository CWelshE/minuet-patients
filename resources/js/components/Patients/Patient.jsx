import React from "react";
import PropTypes from "prop-types";

// Represents a single patient in a PatientsList. Presentational
// component only.
const Patient = (props) => (
  <div className={"patient"}>
    <h3>
      {props.firstName} {props.lastName}
    </h3>
    <span>Date of Birth: {props.dob}</span>
    <span>Phone Number: {props.phone}</span>
  </div>
);

Patient.propTypes = {
  firstName: PropTypes.string.isRequired,
  lastName: PropTypes.string.isRequired,
  dob: PropTypes.string.isRequired,
  phone: PropTypes.string.isRequired,
};

export default Patient;
