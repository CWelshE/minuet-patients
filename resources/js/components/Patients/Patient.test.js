import React from "react";
import renderer from "react-test-renderer";
import Patient from "./Patient.jsx";
import { configure, shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

// Start Enzyme adapter for React
configure({ adapter: new Adapter() });

const patientProps = {
  firstName: "John",
  lastName: "Doe",
  dob: "06-12-1985",
  phone: "123-456-7890",
  handleClick: () => {},
};

it("renders equally compared to snapshot", () => {
  const tree = renderer.create(<Patient {...patientProps} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("has patient info from props", () => {
  const patient = renderer.create(<Patient {...patientProps} />).toJSON();
  expect(patient.children[0].children).toStrictEqual(["John", " ", "Doe"]);
});
