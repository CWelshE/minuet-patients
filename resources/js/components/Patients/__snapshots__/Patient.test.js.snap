// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders equally compared to snapshot 1`] = `
<div
  className="patient css-ssppsg-Patient"
  onClick={[Function]}
>
  <h3>
    John
     
    Doe
  </h3>
  <span>
    Date of Birth: 
    06-12-1985
  </span>
  <span>
    Phone Number: 
    123-456-7890
  </span>
</div>
`;
